backtrace-library/src/main/java/backtraceio/library/common/ApplicationMetadataCache.java

```diff
@@ -6,44 +6,68 @@
 
 import backtraceio.library.logger.BacktraceLogger;
 
-public class ApplicationHelper {
-    private static final transient String LOG_TAG = ApplicationHelper.class.getSimpleName();
+public class ApplicationMetadataCache {
+
+    private static final transient String LOG_TAG = ApplicationMetadataCache.class.getSimpleName();
+
+    private static ApplicationMetadataCache instance;
+
     /**
      * Cached application name
      */
-    private static String applicationName;
+    private String applicationName;
 
     /**
      * Cached application version
      */
-    private static String applicationVersion;
+    private String applicationVersion;
 
     /**
      * Cached package name
      */
-    private static String packageName;
+    private String packageName;
+
+    /**
+     * Returns current application cache. This instance is a singleton since we can only operate
+     * in a single application scope.
+     *
+     * @param context Application context
+     * @return Application metadata cache
+     */
+    public static ApplicationMetadataCache getInstance(Context context) {
+        if (instance == null) {
+            instance = new ApplicationMetadataCache(context);
+        }
+        return instance;
```

Comment:
should we worry about thread safety?

Reply:
It shouldn't be a problem in this case, however lets make sure nothing bad will happen. Adding "double checked lock in pattern"